modules/internal/DefaultRectangular.chpl

```diff
@@ -1761,6 +1761,7 @@ module DefaultRectangular {
       // that along with the size of the array. This is only possible when the
       // byte order is set to native or its equivalent.
       pragma "no prototype"
+      pragma "fn synchronization free"
       extern proc sizeof(type x): size_t;
```

Comment:
Can we just switch to using `c_sizeof` instead of? 

Probably for a different PR, but can we remove the "no prototype" pragma? Does it do anything extra that `extern proc` doesn't?